app-modules/student-data-model/src/Filament/Resources/StudentResource/RelationManagers/EngagementsRelationManager.php

```diff
@@ -190,9 +203,13 @@ public function table(Table $table): Table
                 TextColumn::make('status')
                     ->getStateUsing(fn (Timeline $record) => match ($record->timelineable::class) {
                         EngagementResponse::class => $record->timelineable->status,
-                        default => ''
+                        Engagement::class => EngagementDisplayStatus::getStatus($record->timelineable)->getLabel(),
```

Comment:
I think you should return the enum without `getLabel()` here, then remove the `color()` call completely. Filament will treat it as it is an enum that was cast by the model.

Reply:
Ah, good point. Done.